File: app/src/main/java/de/baumann/browser/unit/IntentUnit.java
package de.baumann.browser.unit;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;

import de.baumann.browser.Ninja.R;

public class IntentUnit {

    public static void share(Context context, String title, String url) {
        Intent sharingIntent = new Intent(Intent.ACTION_SEND);
        sharingIntent.setType("text/plain");
        sharingIntent.putExtra(Intent.EXTRA_SUBJECT, title);
        sharingIntent.putExtra(Intent.EXTRA_TEXT, url);
        context.startActivity(Intent.createChooser(sharingIntent, (context.getString(R.string.menu_share_link))));
    }

    // activity holder
    @SuppressLint("StaticFieldLeak")
    private static Context context = null;
    public static void setContext(Context holder) {
        context = holder;
    }
    public static Context getContext() {
        return context;
    }
}

File: app/src/main/java/de/baumann/browser/unit/IntentUnit.kt
package de.baumann.browser.unit

import android.content.Intent
import de.baumann.browser.Ninja.R
import android.annotation.SuppressLint
import android.content.Context
import de.baumann.browser.unit.IntentUnit

object IntentUnit {
    fun share(context: Context, title: String?, url: String?) {
        val sharingIntent = Intent(Intent.ACTION_SEND)
        sharingIntent.type = "text/plain"
        sharingIntent.putExtra(Intent.EXTRA_SUBJECT, title)
        sharingIntent.putExtra(Intent.EXTRA_TEXT, url)
        context.startActivity(Intent.createChooser(sharingIntent, context.getString(R.string.menu_share_link)))
    }

    // activity holder
    @JvmStatic
    @SuppressLint("StaticFieldLeak")
    var context: Context? = null
}